lightningd/log.c

```diff
@@ -564,27 +565,39 @@ void logv(struct logger *log, enum log_level level,
 {
 	int save_errno = errno;
 	struct log_entry *l = new_log_entry(log, level, node_id);
+	char *log_msg = NULL;
 
 	/* This is WARN_UNUSED_RESULT, because everyone should somehow deal
 	 * with OOM, even though nobody does. */
-	if (vasprintf(&l->log, fmt, ap) == -1)
+	if (vasprintf(&log_msg, fmt, ap) == -1)
 		abort();
 
-	size_t log_len = strlen(l->log);
+	char **lines = tal_strsplit(
+	    log, json_escape_unescape(log, (struct json_escape *)log_msg), "\n",
+	    STR_NO_EMPTY);
```

Comment:
Cute hack.  Minor issue is that other control characters (\b, \r etc) will now get replaced with '?' but they're less common.  \" being turned back into " is probably a win, to be fair...

Reply:
Thanks for the feedback! Should I add a comment at the top about this behavior if needed?

Reply:
You could, but at least note it in the commit message?